model/validation.go

```diff
@@ -204,16 +204,353 @@ func (t Tags) Validate() error {
 	return errs
 }
 
+func validateFQDN(fqdn string) error {
```

Comment:
It might be useful to list all errors rather than the first one. You can use `multierror.Append` to aggregate all errors, see the usage in other validation methods.